tests/test_feature_names.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+import pandas as pd
+import tests.helpers as th
+from numpy.testing import assert_array_equal
+import sklearn
+from sklearn.impute import SimpleImputer
+from sklearn.preprocessing import StandardScaler
+from sklearn.pipeline import Pipeline
+from sklearn.compose import ColumnTransformer
+import category_encoders as encoders
+from unittest import TestCase
+
+__author__ = 'JaimeArboleda'
+
+# data definitions
+X = th.create_dataset(n_rows=100)
+cat_columns = ['categorical', 'na_categorical']
+num_columns = ['float']
+X = X[cat_columns + num_columns]
+np_y = np.random.randn(X.shape[0]) > 0.5
+y = pd.DataFrame(np_y)
+
+class TestEncodersFeaturesOut(TestCase):
+
+    def test_feature_names_out(self):
+        for encoder_name in encoders.__all__:
+            if sklearn.__version__ < "1.2.0":
+                continue
```

Comment:
will the test fail with sklean < 1.2?  
if so I'd force sklearn >= 1.2.0 in the requirement.txt and setup.py

Reply:
Yes, some tests will fail, because the columns obtained in `get_feature_names_out` won't match the names in the output. But I preferred to put this condition just in case some developer want to run the tests and does not have the latest version of `sklearn`. At least they won't crash.

Reply:
that's fine